### PR TITLE
storage: remove asynchronous snapshot queues

### LINKED ISSUE #9588: storage: remove RaftTransport "snapshot" queues
Snapshots are always sent via `RaftTransport.SendSnapshot` making the "snapshot" queues for `RaftTransport.SendAsync` unnecessary/unused.

Cc @cockroachdb/stability

### PR DESCRIPTION
**Pull request by tamird**
Fixes #9588. Spiritually reverts e4be3d8.

---

This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/9590)

### CODE REVIEW

File: storage/raft_transport.go [hunk 1]
```diff
@@ -541,7 +538,9 @@ func (t *RaftTransport) SendAsync(req *RaftMessageRequest) bool {
 		// needs an explicit range ID.
 		panic("only heartbeat messages may be sent to range ID 0")
 	}
-	isSnap := req.Message.Type == raftpb.MsgSnap
+	if isSnap := req.Message.Type == raftpb.MsgSnap; isSnap {
+		panic("snapshots may not be sent asynchronously")
```
Comment (petermattis) [inline storage/raft_transport.go]: Might be better to tell the reader what to do: `snapshots must be sent using SendSnapshot`.

`s/isSnap := req.Message.Type == raftpb.MsgSnap; isSnap/req.Message.Type == raftpb.MsgSnap/g`

**Review by petermattis — APPROVED** [review]
LGTM

**Comment by tamird** [comment]
Review status: 0 of 3 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[storage/raft_transport.go, line 542 at r1](https://reviewable.io:443/reviews/cockroachdb/cockroach/9590#-KSl7e1tJe4SDhuwMEsA-r1-542:-KSl9hD9WhcxbADi5Kb4:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/81f5483e65407b3a29b53f8eec432beea93a7326/storage/raft_transport.go#L542)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> Might be better to tell the reader what to do: `snapshots must be sent using SendSnapshot`.
> 
> 
> 
> `s/isSnap := req.Message.Type == raftpb.MsgSnap; isSnap/req.Message.Type == raftpb.MsgSnap/g`
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9590)*

**Comment by jordanlewis** [comment]
:lgtm:

---

Reviewed 1 of 3 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9590#-:-KSlAI7XIUYnYi8DAfO6:bnfp4nl)*

**Comment by petermattis** [comment]
I think there is still cruft related to snapshots in the `SendAsync` code path.  For example, `RaftTransport.processQueue` has references to `MsgSnap`. Is `RaftTransport.SnapshotStatusChan` used now?

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9590#-:-KSlAOCu94ycz2JjlEsH:bnetbe0)*

**Comment by tamird** [comment]
Removed more cruft.

**Comment by petermattis** [comment]
:lgtm:

---

Review status: 1 of 4 files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[storage/store.go, line 2870 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/9590#-KSlDB0tHSQNOy1eIbhF:-KSlDB0tHSQNOy1eIbhG:b-w7khq9) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd8df4a31bd7409555fe9c0f8155d5d1993c8dce/storage/store.go#L2870)):*
> ```Go
> 					// replica has been removed, Replica.mu.destroyed will be non-nil
> 					// causing Replica.reportSnapshotStatus() to be a no-op.
> 					r.reportSnapshotStatus(st.Req.Message.To, st.Err)
> ```

`Replica.reportSnapshotStatus` will be unused after this removal. We need to fix that (but it can wait to a follow-on PR). See #9592.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9590#-:-KSlDMwQOmJC-BLVxj9D:bnfp4nl)*

**Comment by tamird** [comment]
Review status: 1 of 4 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[storage/store.go, line 2870 at r3](https://reviewable.io:443/reviews/cockroachdb/cockroach/9590#-KSlDB0tHSQNOy1eIbhF:-KSlDf1oVMnUp9-GQwya:b-tdsi29) ([raw file](https://github.com/cockroachdb/cockroach/blob/bd8df4a31bd7409555fe9c0f8155d5d1993c8dce/storage/store.go#L2870)):*
<details><summary><i>Previously, petermattis (Peter Mattis) wrote…</i></summary>
> `Replica.reportSnapshotStatus` will be unused after this removal. We need to fix that (but it can wait to a follow-on PR). See #9592.
</details>
Well, it's already unused :)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9590)*

**Pull request merged**